app/events/rtjc_test.go

```diff
@@ -34,34 +33,139 @@ func TestRtjc_isPinned(t *testing.T) {
 	}
 }
 
-func TestRtjc_summary(t *testing.T) {
-	oai := &mocks.OpenAISummary{
-		SummaryFunc: func(text string) (string, error) {
-			return "ai summary", nil
-		},
+func makeRtjc(submitter *mocks.Submitter, summarizer *mocks.Summarizer) Rtjc {
+	return Rtjc{
+		Port:            1,
+		Submitter:       submitter,
+		Summarizer:      summarizer,
+		Swg:             syncs.NewSizedGroup(1),
+		SubmitRateLimit: 1,
+		SubmitRateBurst: 100,
+	}
+}
+
+func TestRtjc_ReadMessage(t *testing.T) {
+	tbl := []struct {
+		name            string
+		input           string
+		callsSubmit     int
+		callsSubmitHTML int
+		callsSummary    int
+	}{
+		{"Begin", "⚠️ Вещание подкаста началось - https://stream.radio-t.com/", 1, 0, 1},
+		{"New theme", "⚠️ blah blah - https://link.example.com", 1, 1, 1},
+		{"Remark", "⚠️ blah blah - https://radio-t.com/p/2023/04/04/prep-853/", 1, 2, 1},
+		{"Blah", "blah blah - https://link.example.com", 1, 0, 0},
+	}
+
+	for _, tt := range tbl {
+		t.Run(tt.name, func(t *testing.T) {
+			sb := &mocks.Submitter{
+				SubmitFunc: func(ctx context.Context, text string, pin bool) error {
+					return nil
+				},
+				SubmitHTMLFunc: func(ctx context.Context, text string, pin bool) error {
+					return nil
+				},
+			}
+
+			sm := &mocks.Summarizer{
+				GetSummariesByMessageFunc: func(link string) (messages []string, err error) {
+					// Begin and end of stream
+					if strings.Contains(link, "stream.radio-t.com") {
+						return []string{}, nil
+					}
+
+					// Remark link
+					if strings.Contains(link, "radio-t.com/p/") {
+						return []string{"summary", "summary2"}, nil
+					}
+
+					// Regular link
+					return []string{"summary"}, nil
+				},
+			}
+
+			wasRead := false
+			io := &mocks.IoReader{
+				ReadFunc: func(p []byte) (n int, err error) {
+					if wasRead {
```

Comment:
i see you actually may need Reader mock as your logic here is non-trivial. However, in this case make mock manually in good-old way, will be a better choice